be/src/olap/tablet.cpp

```diff
@@ -938,6 +938,38 @@ Status Tablet::capture_rs_readers(const Version& spec_version, std::vector<RowSe
     return Status::OK();
 }
 
+Status Tablet::capture_sub_txn_rs_readers(int64_t version, const std::vector<int64_t>& sub_txn_ids,
+                                          std::vector<RowSetSplits>* rs_splits) {
+    LOG(INFO) << "capture_sub_txn_rs_readers for partition=" << partition_id()
+              << ", tablet_id=" << tablet_id() << ", version=" << version
+              << ", sub_txn_ids.size=" << sub_txn_ids.size();
+    for (int i = 0; i < sub_txn_ids.size(); ++i) {
+        auto sub_txn_id = sub_txn_ids[i];
+        auto rowset = _engine.txn_manager()->get_tablet_rowset(tablet_id(), tablet_uid(),
+                                                               partition_id(), sub_txn_id);
+        DCHECK(rowset != nullptr) << " rowset is nullptr for sub_txn_id=" << sub_txn_ids[i]
+                                  << ", partition_id=" << partition_id()
+                                  << ", tablet=" << tablet_id();
+        int64_t tmp_version = version + i + 1;
```

Comment:
version has already been added in prallel_scanner_builder _build_scanners_by_rowid.

Reply:
_load is called firstly, now version is not changed